tensorflow/java/src/test/java/org/tensorflow/TensorTest.java

```diff
@@ -105,55 +116,90 @@ public void createFromBufferWithNonNativeByteOrder() {
   }
 
   @Test
-  public void createWithTypedBuffer() {
-    int[] ints = {1, 2, 3, 4};
-    float[] floats = {1f, 2f, 3f, 4f};
-    double[] doubles = {1d, 2d, 3d, 4d};
-    long[] longs = {1L, 2L, 3L, 4L};
-    long[] shape = {4};
+  public void whenCreateUsingDoubleBufferThenExpectedMatrixIsStoredInTensor() {
+    final double[] expected = {1d, 2d, 3d, 4d};
+    final long[] shape = {4};
+
+    try (final Tensor t = Tensor.create(shape, DoubleBuffer.wrap(expected))) {
+      final double[] actual = new double[expected.length];
+      t.copyTo(actual);
+      assertArrayEquals(expected, actual, EPSILON);
```

Comment:
My reasoning behind the three lines was:

- I prefer to break tests up so that the arrange/act/assert statements are separate where possible (as it improves readability which is the most important aspect of tests)
- Most junit asserts, like this one, are backwards compared to assertThat, and I wanted to be absolutely sure I got them in the right order.

It can change if that is preferred, but I don't think it should.